### ISSUE TITLE
GFF3 loader error during import of protein

### ISSUE THREAD

**Issue by risharde** [tripal/tripal#1298]
### Issue description
When importing a GFF3 file with proteins, the GFF3 loader is unable to complete since it throws an error.

#### Steps to reproduce
Create organism: Arabidopsis thaliana
Import Sequence Ontology (if not already imported)
Import the following GFF3 file: https://www.arabidopsis.org/download_files/Genes/TAIR10_genome_release/TAIR10_gff3/TAIR10_GFF3_genes.gff
Landmark Type: chromosome
Run the GFF loader

```
This error message is actually taken from Tripal 4 GFF3 loader but it should be similar for Tripal 3:
ERROR: SQLSTATE[23502]: Not null violation: 7 ERROR:  null value in column "type_id" violates not-null constraint
DETAIL:  Failing row contains (593060, null, 2, AT1G01010.1,AT1G01010.1-Protein, AT1G01010.1,AT1G01010.1-Protein-protein, , 0,                                 , null, f, f, 2022-09-11 01:31:13.563705, 2022-09-11 01:31:13.563705).: 
      INSERT INTO "feature"
        (uniquename, name, type_id, organism_id, residues, md5checksum,
         seqlen, is_analysis, is_obsolete)
      VALUES
(:uniquename_1, :name_1, :type_id_1, :organism_id_1, :residues_1,  :md5checksum_1, :seqlen_1, FALSE, FALSE),
```

Tagging @spficklin

**Comment by dsenalik** [comment]
I'm working on loading a new genome now, so I though I would investigate this. Using one short gene from that gff3
```
Chr1	TAIR10	gene	11649	13714	.	-	.	ID=AT1G01030;Note=protein_coding_gene;Name=AT1G01030
Chr1	TAIR10	mRNA	11649	13714	.	-	.	ID=AT1G01030.1;Parent=AT1G01030;Name=AT1G01030.1;Index=1
Chr1	TAIR10	protein	11864	12940	.	-	.	ID=AT1G01030.1-Protein;Name=AT1G01030.1;Derives_from=AT1G01030.1
Chr1	TAIR10	five_prime_UTR	13335	13714	.	-	.	Parent=AT1G01030.1
Chr1	TAIR10	exon	13335	13714	.	-	.	Parent=AT1G01030.1
Chr1	TAIR10	five_prime_UTR	12941	13173	.	-	.	Parent=AT1G01030.1
Chr1	TAIR10	CDS	11864	12940	.	-	0	Parent=AT1G01030.1,AT1G01030.1-Protein;
Chr1	TAIR10	three_prime_UTR	11649	11863	.	-	.	Parent=AT1G01030.1
```
I see this in the postgres log, the last set of values is the problem
```
INSERT INTO chado.feature
  (uniquename,                                      name,                              type_id, organism_id, residues, md5checksum, seqlen, is_analysis, is_obsolete)
VALUES
  ('AT1G01030',                                     'AT1G01030',                       '499',   '1', '',  '', '0', FALSE, FALSE),
  ('AT1G01030.1',                                   'AT1G01030.1',                     '425',   '1', '',  '', '0', FALSE, FALSE),
  ('AT1G01030.1-Protein',                           'AT1G01030.1',                     '534',   '1', '',  '', '0', FALSE, FALSE),
  ('AT1G01030.1-five_prime_utr-Chr1:13335..13714',  'AT1G01030.1-five_prime_utr',      '585',   '1', '',  '', '0', FALSE, FALSE),
  ('AT1G01030.1-exon-Chr1:13335..13714',            'AT1G01030.1-exon',                '596',   '1', '',  '', '0', FALSE, FALSE),
  ('AT1G01030.1-five_prime_utr-Chr1:12941..13173',  'AT1G01030.1-five_prime_utr',      '585',   '1', '',  '', '0', FALSE, FALSE),
  ('AT1G01030.1-cds-Chr1:11864..12940',             'AT1G01030.1-cds',                 '535',   '1', '',  '', '0', FALSE, FALSE),
  ('AT1G01030.1-three_prime_utr-Chr1:11649..11863', 'AT1G01030.1-three_prime_utr',     '668',   '1', '',  '', '0', FALSE, FALSE),
  ('AT1G01030.1,AT1G01030.1-Protein-protein',       'AT1G01030.1,AT1G01030.1-Protein', NULL,    '1', '',  '', '0', FALSE, FALSE)
```

The line that is a problem because it has two parents is
```
Chr1     TAIR10  CDS     11864   12940   .       -       0       Parent=AT1G01030.1,AT1G01030.1-Protein;
```

**Comment by dsenalik** [comment]
Actually the error was because the cv term for ```polypeptide``` didn't get looked up, add code to make sure it gets looked up, and if not give a useful error message. Of course the double parent problem means the name is nonsense, so that can be fixed too.
Another issue is that the file uses xxx-Protein and the loader uses xxx-protein, so a regex is needed too unless you disable the automatic protein generation.
![20220928_protein](https://user-images.githubusercontent.com/8419404/192852433-5474815f-b00a-4adc-a9e6-f59fb2defeec.png)

**Comment by dsenalik** [comment]
To note here in the issue one additional problem solved:
If the gff3 has a ```Derives_from=xxx``` annotation, and that ```xxx``` value does not appear in the gff3, meaning we have no record for the parent feature, it will now be ignored during ```Step 24 of 27: Insert 'derives_from' relationships...```

**Comment by dsenalik** [comment]
This additional problem solved is actually Issue #1299 so pull request #1305 will fix that too.

**Comment by dsenalik** [comment]
Closed with merge of #1305
